install_microshift.py

```diff
@@ -0,0 +1,178 @@
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cmd = f"sudo composer-cli blueprints push minimal-microshift.toml"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cmd = f"sudo composer-cli blueprints depsolve minimal-microshift | grep microshift"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    compose_command = f"sudo composer-cli compose start-ostree --ref 'rhel/9/{uname_m}/edge' minimal-microshift edge-container"
+
+    compose_output = h.run(compose_command)
+
+    build_id = compose_output.out.split(" ")[1]
+
+    logger.info(build_id)
+
+    while True:
+        status_output = h.run(f"sudo composer-cli compose status").out.strip()
+        if 'FINISHED' in status_output:
+            break
+        logger.info("waiting on build to finish")
+        time.sleep(5)
+
+    cmd = f"sudo composer-cli compose image {build_id}"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    uid = os.getuid()
+    gid = os.getgid()
+
+    os.chown(f"{build_id}-container.tar", uid, gid)
+
+    os.chmod(f"{build_id}-container.tar", 444)
+
+    image_id_command = f"sudo podman load < ./{build_id}-container.tar"
+
+    result = h.run(image_id_command).out
+
+    image_id = result.split(':')[-1].strip()
+
+    cmd = f"sudo podman run -d --name=minimal-microshift-server -p 8085:8080 {image_id}"
+    result = h.run(cmd)
+
+    cmd = f"sudo composer-cli blueprints push microshift-installer.toml"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    url = "http://127.0.0.1:8085/repo/"
+    ref = f"rhel/9/{uname_m}/edge"
+
+    build_command_two = f"sudo composer-cli compose start-ostree --url {url} --ref \"{ref}\" microshift-installer edge-installer"
+    result= h.run(build_command_two)
+
+    build_id_two = result.out.split(" ")[1]
+
+    time.sleep(10)
+
+    while True:
+        status_output = h.run(f"sudo composer-cli compose status").out
+
+        build_status_line = next((line for line in status_output.split('\n') if build_id_two in line), None)
+
+        if 'FINISHED' in build_status_line:
+            break
+
+        logger.info("Waiting on build to finish")
+        time.sleep(2)
+
+    cmd = f"sudo composer-cli compose image {build_id_two}"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    os.chown(f"{build_id_two}-installer.iso", uid, gid)
+
+    os.chmod(f"{build_id_two}-installer.iso", 444)
+
+    vgname_command = "vgs --noheadings -o vg_name"
+    vgname_output = h.run(vgname_command)
+    vgname = vgname_output.out.split(" ")[0]
+
+    cmd = f"sudo vgrename {vgname} rhel"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cmd = f"sudo yum install -y lorax"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    cmd = f"sudo mkksiso kickstart.ks {build_id_two}-installer.iso final.iso"
+    output = h.run(cmd)
+    logger.info(output.out)
+
+    logger.info("Microshift iso is ready in final.iso")
+
+    os.remove("vim /var/cache/osbuild-worker/osbuild-store/stage/") 
```

Comment:
Did you test this? This runs VIM....

Please make sure to have things tested.

Reply:
I tested the code and it does work. I think when making some corrections I must have made some mistakes however. Apologies for that.